devito/passes/iet/dtypes.py

```diff
@@ -1,14 +1,54 @@
 import numpy as np
 import ctypes
 
-from devito.ir import FindSymbols, Uxreplace
+from devito.ir import FindSymbols
+from devito.ir.iet.nodes import Dereference
+from devito.ir.iet.visitors import Uxreplace
+from devito.symbolics.extended_dtypes import Float16P
+from devito.tools.utils import as_list
+from devito.types.basic import Symbol
 
-__all__ = ['lower_complex']
+__all__ = ['lower_dtypes']
 
 
-def lower_complex(iet, lang, compiler):
+def lower_dtypes(iet, lang, compiler, sregistry):
```

Comment:
how can this not be an `@iet_pass` and still work?

Reply:
It's called from an `@iet_pass` [in `definitions` here](https://github.com/devitocodes/devito/blob/516b4ad0adc0f9eca18f5bd072a80051557cccf4/devito/passes/iet/definitions.py#L416)

Reply:
ah OK , I maybe have told Mathias already, but imho that thing goes straight into `operator/operator.py::Operator::_lower_iet`

Reply:
Ah okay, I can make this change